plugins/de.cau.cs.kieler.klighd.lsp/src/de/cau/cs/kieler/klighd/lsp/utils/RenderingPreparer.xtend

```diff
@@ -0,0 +1,442 @@
+        } else {
+            // In this case the element is a KEdge.
+            bounds = edgeBounds(element as KEdge)
+        }
+        // Decide if the bounds should be put in the boundsMap or in the rendering's properties.
+        if (boundsMap === null) {
+            rendering.setBounds(bounds)
+        } else {
+            boundsMap.put(rendering.id, bounds)
+        }
+        // Calculate the bounds and decorations of all child renderings.
+        if (rendering instanceof KContainerRendering) {
+            handleChildren(rendering.children, rendering.childPlacement, bounds, boundsMap, decorationMap, element)
+        }
+        // Calculate the bounds for the junction point rendering.
+        if (rendering instanceof KPolyline) {
+            if (rendering.junctionPointRendering !== null) {                
+                handleAreaAndPointAndDecoratorPlacementRendering(rendering.junctionPointRendering, bounds, boundsMap,
+                    decorationMap, element)
+            }
+        }
+    }
+    
+    /**
+     * Calculate the size and position of all child renderings recursively. The boundsMap and decorationMap again indicate
+     * if is should be stored in them (not null) or in the rendering's properties themselves.
+     * Inspired by {@link de.cau.cs.kieler.klighd.piccolo.internal.controller.AbstractKGERenderingController}.
+     * 
+     * @param renderings The child renderings to calculate the sizes and decorations for.
+     * @param placement The defined placement of the child renderings.
+     * @param parentBounds The bounds of the parent rendering. All child renderings have to fit inside these.
+     * @param boundsMap The map to store the bounds in. If it is null, the bounds should be instead stored in the
+     * properties of the {@code rendering} itself.
+     * @param decorationMap The map to store the decoration in. If it is null, the decoration should be instead stored in the
+     * properties of the {@code rendering} itself.
+     * @param parent The parent graph element containing this rendering as one of its children.
+     */
+    private static def void handleChildren(List<KRendering> renderings, KPlacement placement, Bounds parentBounds,
+        Map<String, Bounds> boundsMap, Map<String, Decoration> decorationMap, KGraphElement parent) {
+        if (placement instanceof KGridPlacement) {
+            handleGridPlacementRendering(renderings, placement, parentBounds, boundsMap, decorationMap, parent)
+        } else {
+            for (rendering : renderings) {
+                handleAreaAndPointAndDecoratorPlacementRendering(rendering, parentBounds, boundsMap, decorationMap, parent)
+            }
+        }
+    }
+    
+    /**
+     * Handles the child renderings, if they should be placed in a grid placement.
+     * 
+     * @see #handleChildren
+     */
+    private static def void handleGridPlacementRendering(List<KRendering> renderings, KGridPlacement gridPlacement, 
```

Comment:
Add param tags

Reply:
I intentionally left them out as I used the @see tag to prevent duplicate and therefore possibly inconsistent documentation.